Description: Upstream changes introduced in version 3.0~a24-1ubuntu13
 .
   * Mimic livecd-rootfs's handling of inodes on ext* filesystems to
     reduce the pain of resizing preinstalled images. (LP: #836810)
 .
   * This patch is a dirty hack, and needs some TLC and generalisation
     before it's reasonably upstreamable.
Author: Adam Conrad <adconrad@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/836810

--- live-build-3.0~a24.orig/scripts/build/lb_binary_rootfs
+++ live-build-3.0~a24/scripts/build/lb_binary_rootfs
@@ -122,7 +122,7 @@ case "${LB_CHROOT_FILESYSTEM}" in
 		case "${LB_BUILD_WITH_CHROOT}" in
 			true)
 				dd if=/dev/zero of=chroot/filesystem.${LB_CHROOT_FILESYSTEM} bs=1024k count=0 seek=${REAL_DIM}
-				Chroot chroot "mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -N ${REAL_INODES} -m 0 filesystem.${LB_CHROOT_FILESYSTEM}"
+				Chroot chroot "mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -i 8192 -m 0 filesystem.${LB_CHROOT_FILESYSTEM}"
 
 				mkdir -p filesystem.tmp
 				${LB_ROOT_COMMAND} mount -o loop chroot/filesystem.${LB_CHROOT_FILESYSTEM} filesystem.tmp
@@ -187,7 +187,7 @@ case "${LB_CHROOT_FILESYSTEM}" in
 
 			false)
 				dd if=/dev/zero of=binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM} bs=1024k count=0 seek=${REAL_DIM}
-				mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -N ${REAL_INODES} -m 0 binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM}
+				mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -i 8192 -m 0 binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM}
 
 				mkdir -p filesystem.tmp
 				${LB_ROOT_COMMAND} mount -o loop binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM} filesystem.tmp
